packages/edit-site/src/posts.js

```diff
@@ -1,24 +1,96 @@
 /**
  * WordPress dependencies
  */
+import { store as blocksStore } from '@wordpress/blocks';
+import {
+	registerCoreBlocks,
+	__experimentalGetCoreBlocks,
+	__experimentalRegisterExperimentalCoreBlocks,
+} from '@wordpress/block-library';
+import { dispatch } from '@wordpress/data';
 import { createRoot, StrictMode } from '@wordpress/element';
+import { store as editorStore } from '@wordpress/editor';
+import { store as preferencesStore } from '@wordpress/preferences';
+import {
+	registerLegacyWidgetBlock,
+	registerWidgetGroupBlock,
+} from '@wordpress/widgets';
 
 /**
  * Internal dependencies
  */
+import './hooks';
+import { store as editSiteStore } from './store';
+
+/**
+ * Internal dependencies
+ */
+// import { initializeEditor } from './index';
 import PostsApp from './components/posts-app';
 
 /**
  * Initializes the "Posts Dashboard"
- * @param {string} id DOM element id.
+ * @param {string} id       ID of the root element to render the screen in.
+ * @param {Object} settings Editor settings.
  */
-export function initializePostsDashboard( id ) {
+export function initializePostsDashboard( id, settings ) {
 	if ( ! globalThis.IS_GUTENBERG_PLUGIN ) {
 		return;
 	}
 	const target = document.getElementById( id );
 	const root = createRoot( target );
 
+	dispatch( blocksStore ).reapplyBlockTypeFilters();
```

Comment:
This code seems to be a duplication of packages/edit-site/src/index.js, could we create some function used in both places to avoid the duplication?
I fear in the future someone may easily change code in a place forgetting about the other place.

Reply:
I think duplication is fine sometimes, as it reminds us that this code is unnecessary boilerplate, it's not clear why we need to call it here in the first place.

Reply:
I guess if we are duplicating at least we should include comments in both files saying they are duplication of each other. Duplication ends up leading to divergence and we may end up with the double of the code to maintain.
I think code like block registration etc may be needed because the preview of the post depends on the blocks being registered and the editor being properly initialized.